### PR TITLE
TDI-37967:Component tXSDValidator returns null in flow variable.

### PR DESCRIPTION
**Pull request by pyzhou-talend**
https://jira.talendforge.org/browse/TDI-37967

### CODE REVIEW

**Comment by jzhao-talend** [comment]
You need to check that whether:
`globalMap.put("<%=cid %>_XSD_ERROR_MESSAGE", "" + errorMessage_<%=cid%>);`
this kind of code is set correct or not

**Comment by pyzhou-talend** [comment]
OK. I have modified the code according to your comment.

File: main/plugins/org.talend.designer.components.localprovider/components/tXSDValidator/tXSDValidator_main.javajet [hunk 1]
```diff
@@ -175,6 +176,7 @@ if("FLOW_MODE".equals(mode)){
 							}
 %>
 						} else {
+						        fullErrorMessage_<%=cid%>.append("For Row "+nb_line_<%=cid%>+" <%=column.getLabel()%>: "+errorMessage_<%=cid%>+'\n');
```
Comment (jzhao-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tXSDValidator/tXSDValidator_main.javajet]: I don't understand why here use a StringBuffer ?
If input have 3 records:
> 1--->reject
> 2--->pass
> 3--->reject
it mean that record 3 would get error message 1+3.

File: main/plugins/org.talend.designer.components.localprovider/components/tXSDValidator/tXSDValidator_main.javajet [hunk 2]
```diff
@@ -137,6 +137,9 @@ if("FLOW_MODE".equals(mode)){
 					<%=rejectConnName %> = new <%=rejectConnName %>Struct();
 <%
 				}
+%>
+				String errorMessageRow_<%=cid%> = "";
```
Comment (jzhao-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tXSDValidator/tXSDValidator_main.javajet]: I prefer to this default value to be null. 
you can check another mode. 
Keep same behavior.

File: main/plugins/org.talend.designer.components.localprovider/components/tXSDValidator/tXSDValidator_main.javajet [hunk 3]
```diff
@@ -209,6 +213,9 @@ if("FLOW_MODE".equals(mode)){
                     }
 	    		}
 	    	}
+%>
+	    	globalMap.put("<%=cid %>_XSD_ERROR_MESSAGE", errorMessageRow_<%=cid%>);
```
Comment (jzhao-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tXSDValidator/tXSDValidator_main.javajet]: Did you check those 2 flow variables whether also need to put values ?
```
globalMap.put("<%=cid %>_DIFFERENCE", "" + validate_<%=cid %>);
globalMap.put("<%=cid %>_VALID", (validate_<%=cid %> == 1)?true:false); 
```

File: main/plugins/org.talend.designer.components.localprovider/components/tXSDValidator/tXSDValidator_main.javajet (hunk 2 above)
Comment (pyzhou-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tXSDValidator/tXSDValidator_main.javajet]: In file mode, It is empty String as well, So I will keep same behavior with it.

File: main/plugins/org.talend.designer.components.localprovider/components/tXSDValidator/tXSDValidator_main.javajet (hunk 3 above)
Comment (pyzhou-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tXSDValidator/tXSDValidator_main.javajet]: I will add this two.

**Comment by pyzhou-talend** [comment]
@jzhao-talend I have modified the code according to your comment.

**Comment by pyzhou-talend** [comment]
thanks~

**Pull request merged**